src/material.h

```diff
@@ -123,48 +130,123 @@ namespace pyne
+    
+    /// Loads the matrial composition from an HDF5 file according to the layout
+    /// defined by protocol 1.  This protocol should be used in favor of protocol 0.
+    /// \param db HDF5 id for the open HDF5 file.
+    /// \param mat_name Path to the base node for the material in a db.
+    /// \param nucpath Path to the base node for nuclide list in a db.
+    /// \param row The index to read out, may be negative.
+    void _load_comp_protocol1(hid_t db, std::string mat_name, std::string nucpath, int row);
+    
     /// Loads a material from an HDF5 file into this object.
     /// \param filename Path on disk to the HDF5 file.
-    /// \param datapath Path to the the material in the file.
+    /// \param mat_name Path to the the material in the file.
     /// \param row The index to read out, may be negative.
     /// \param protocol Flag for layout of material on disk.
-    void from_hdf5(char * filename, char * datapath, int row=-1, int protocol=1);
+    void from_hdf5(char * filename, char * mat_name, int row=-1, int protocol=1);
 
     /// Loads a material from an HDF5 file into this object.
     /// \param filename Path on disk to the HDF5 file.
-    /// \param datapath Path to the the material in the file.
+    /// \param mat_name Path to the the material in the file.
     /// \param row The index to read out, may be negative.
     /// \param protocol Flag for layout of material on disk.
-    void from_hdf5(std::string filename, std::string datapath="/material",
+    void from_hdf5(std::string filename, std::string mat_name="/material",
                                                           int row=-1, int protocol=1);
+   
+  private:
+
+    /// Detect the HDF5 file assuming protocol1.
+    /// \param db HDF5 id for the open HDF5 file.
+    /// \param mat_name Path to look for the material in the file.
+    /// Return options are: 
+    ///     -"-1": mat_name and "/material" do not exist
+    ///     - "0": mat_name and/or "/material" exist but either as a group or a dataset
+    ///     - "1": mat_name exists as a dataset -> old layout
+    ///     - "2": "/material" exists as a group-> new layout
+    int detect_hdf5_layout(hid_t db, std::string mat_name);
+  
+    enum prot1_layout {path_donotexists=-1, unknown, old_layout, new_layout};
+  
+  public:
 
     /// Writes this material out to an HDF5 file.
     /// This happens according to protocol 1.
     /// \param filename Path on disk to the HDF5 file.
+    /// \param mat_name Path to the the material in the file.
+    /// \param row The index to read out, may be negative. Also note that this is a
+    ///            float.  A value of -0.0 indicates that the material should be
+    ///            appended to the end of the dataset.
+    /// \param chunksize The chunksize for all material data on disk.
+    /// New write_hdf5 which fallback on the old one when required
+    void write_hdf5(std::string filename, std::string mat_name="/material", 
```

Comment:
I think it was this string "/material" that I was recommending changing to "/mat_name", not the variable name `datapath`.

With this default, you get a group "/material/material" by default which can be a little confusing.

Reply:
I change in most of the places... missed that one... :)

Reply:
ok I did not change the default values.... but all the variables name.....

Reply:
I actually think the variable names make more sense as `datapath`....  And we still call it `datapath` in the python layer